### PR TITLE
Add documentation for Sharepoint connector

### PR DESCRIPTION
**Pull request by aashikam**
## Purpose
$title

### CODE REVIEW

File: en/docs/assets/img/integrate/connectors/sharepoint-store.png
Comment (arunans23) [inline en/docs/assets/img/integrate/connectors/sharepoint-store.png:1]: Shall we use this icon mentioned in this PR? https://github.com/wso2-extensions/esb-connector-sharepoint/pull/25/ This is the icon that will be visible in the connector store.

File: en/docs/reference/connectors/sharepoint-connector/sharepoint-configuration.md [hunk 1]
```diff
@@ -0,0 +1,945 @@
...
+    ```json
+    {
+        "site_id": "site123",
+        "parent_id": "root",
+        "folder_name": "New Folder",
+        "MicrosoftGraphConflictBehavior": "fail"
+    }
+    ```
+
+??? note uploadFile
```
Comment (DinithiDiaz) [inline en/docs/reference/connectors/sharepoint-connector/sharepoint-configuration.md:117]:
```suggestion
??? note "uploadFile"
```

File: en/docs/reference/connectors/sharepoint-connector/sharepoint-configuration.md [hunk 2]
```diff
@@ -0,0 +1,945 @@
...
+
+    ```json
+    {
+        "siteId": "site123",
+        "parentItemId": "root",
+        "fileName": "NewDocument.docx"
+    }
+    ```
+
+??? note createGroup
```
Comment (DinithiDiaz) [inline en/docs/reference/connectors/sharepoint-connector/sharepoint-configuration.md:162]:
```suggestion
??? note "createGroup"
```

File: en/docs/reference/connectors/sharepoint-connector/sharepoint-configuration.md [hunk 3]
```diff
@@ -0,0 +1,945 @@
...
+        "displayName": "Project Alpha Team",
+        "groupTypes": ["Unified"],
+        "mailEnabled": "true",
+        "mailNickname": "projectalpha",
+        "securityEnabled": "false",
+        "visibility": "Private"
+    }
+    ```
+
+??? note createList
```
Comment (DinithiDiaz) [inline en/docs/reference/connectors/sharepoint-connector/sharepoint-configuration.md:235]:
```suggestion
??? note "createList"
```

File: en/docs/reference/connectors/sharepoint-connector/sharepoint-configuration.md [hunk 4]
```diff
@@ -0,0 +1,945 @@
...
+                }
+            ],
+            "list": {
+                "template": "genericList"
+            }
+        }
+    }
+    ```
+
+??? note createListItem
```
Comment (DinithiDiaz) [inline en/docs/reference/connectors/sharepoint-connector/sharepoint-configuration.md:288]:
```suggestion
??? note "createListItem"
```

File: en/docs/reference/connectors/sharepoint-connector/sharepoint-configuration.md [hunk 5]
```diff
@@ -0,0 +1,945 @@
...
+        "siteId": "site123",
+        "listId": "list456",
+        "fields": {
+            "BookAuthor": "Ross Geller",
+            "PageCount": 567
+        }
+    }
+    ```
+
+??? note deleteDriveItem
```
Comment (DinithiDiaz) [inline en/docs/reference/connectors/sharepoint-connector/sharepoint-configuration.md:336]:
```suggestion
??? note "deleteDriveItem"
```

File: en/docs/reference/connectors/sharepoint-connector/sharepoint-configuration.md [hunk 6]
```diff
@@ -0,0 +1,945 @@
...
+    **Sample request**
+
+    ```json
+    {
+        "siteId": "site123",
+        "itemId": "item789"
+    }
+    ```
+
+??? note deleteList
```
Comment (DinithiDiaz) [inline en/docs/reference/connectors/sharepoint-connector/sharepoint-configuration.md:374]:
```suggestion
??? note "deleteList"
```

File: en/docs/reference/connectors/sharepoint-connector/sharepoint-configuration.md [hunk 7]
```diff
@@ -0,0 +1,945 @@
...
+    **Sample request**
+
+    ```json
+    {
+        "siteId": "site123",
+        "listId": "list456"
+    }
+    ```
+
+??? note deleteListItem
```
Comment (DinithiDiaz) [inline en/docs/reference/connectors/sharepoint-connector/sharepoint-configuration.md:412]:
```suggestion
??? note "deleteListItem"
```

File: en/docs/reference/connectors/sharepoint-connector/sharepoint-configuration.md [hunk 8]
```diff
@@ -0,0 +1,945 @@
...
+
+    ```json
+    {
+        "siteId": "site123",
+        "listId": "list456",
+        "itemId": "5"
+    }
+    ```
+
+??? note getDriveItemById
```
Comment (DinithiDiaz) [inline en/docs/reference/connectors/sharepoint-connector/sharepoint-configuration.md:457]:
```suggestion
??? note "getDriveItemById"
```

File: en/docs/reference/connectors/sharepoint-connector/sharepoint-configuration.md [hunk 9]
```diff
@@ -0,0 +1,945 @@
...
+    **Sample request**
+
+    ```json
+    {
+        "siteId": "site123",
+        "itemId": "item789"
+    }
+    ```
+
+??? note getFolderChildren
```
Comment (DinithiDiaz) [inline en/docs/reference/connectors/sharepoint-connector/sharepoint-configuration.md:495]:
```suggestion
??? note "getFolderChildren"
```

File: en/docs/reference/connectors/sharepoint-connector/sharepoint-configuration.md [hunk 10]
```diff
@@ -0,0 +1,945 @@
...
+    **Sample request**
+
+    ```json
+    {
+        "siteId": "site123",
+        "itemId": "folder456"
+    }
+    ```
+
+??? note getGroupSite
```
Comment (DinithiDiaz) [inline en/docs/reference/connectors/sharepoint-connector/sharepoint-configuration.md:533]:
```suggestion
??? note "getGroupSite"
```

File: en/docs/reference/connectors/sharepoint-connector/sharepoint-configuration.md [hunk 11]
```diff
@@ -0,0 +1,945 @@
...
+ 
+    **Sample request**
+
+    ```json
+    {
+        "groupId": "group123"
+    }
+    ```
+
+??? note getListById
```
Comment (DinithiDiaz) [inline en/docs/reference/connectors/sharepoint-connector/sharepoint-configuration.md:564]:
```suggestion
??? note "getListById"
```

File: en/docs/reference/connectors/sharepoint-connector/sharepoint-configuration.md [hunk 12]
```diff
@@ -0,0 +1,945 @@
...
+    **Sample request**
+
+    ```json
+    {
+        "siteId": "site123",
+        "listId": "3fa85f64-5717-4562-b3fc-2c963f66afa6"
+    }
+    ```
+
+??? note getListItemById
```
Comment (DinithiDiaz) [inline en/docs/reference/connectors/sharepoint-connector/sharepoint-configuration.md:602]:
```suggestion
??? note "getListItemById"
```

File: en/docs/reference/connectors/sharepoint-connector/sharepoint-configuration.md [hunk 13]
```diff
@@ -0,0 +1,945 @@
...
+
+    ```json
+    {
+        "siteId": "site123",
+        "listId": "3fa85f64-5717-4562-b3fc-2c963f66afa6",
+        "itemId": "item789"
+    }
+    ```
+
+??? note getListItems
```
Comment (DinithiDiaz) [inline en/docs/reference/connectors/sharepoint-connector/sharepoint-configuration.md:647]:
```suggestion
??? note "getListItems"
```

File: en/docs/reference/connectors/sharepoint-connector/sharepoint-configuration.md [hunk 14]
```diff
@@ -0,0 +1,945 @@
...
+    **Sample request**
+
+    ```json
+    {
+        "siteId": "site123",
+        "listId": "3fa85f64-5717-4562-b3fc-2c963f66afa6"
+    }
+    ```
+
+??? note getLists
```
Comment (DinithiDiaz) [inline en/docs/reference/connectors/sharepoint-connector/sharepoint-configuration.md:685]:
```suggestion
??? note "getLists"
```

File: en/docs/reference/connectors/sharepoint-connector/sharepoint-configuration.md [hunk 15]
```diff
@@ -0,0 +1,945 @@
...
+ 
+    **Sample request**
+
+    ```json
+    {
+        "siteId": "site123"
+    }
+    ```
+
+??? note getRootChildren
```
Comment (DinithiDiaz) [inline en/docs/reference/connectors/sharepoint-connector/sharepoint-configuration.md:716]:
```suggestion
??? note "getRootChildren"
```

File: en/docs/reference/connectors/sharepoint-connector/sharepoint-configuration.md [hunk 16]
```diff
@@ -0,0 +1,945 @@
...
+ 
+    **Sample request**
+
+    ```json
+    {
+        "siteId": "site123"
+    }
+    ```
+
+??? note updateFileContent
```
Comment (DinithiDiaz) [inline en/docs/reference/connectors/sharepoint-connector/sharepoint-configuration.md:747]:
```suggestion
??? note "updateFileContent"
```

File: en/docs/reference/connectors/sharepoint-connector/sharepoint-configuration.md [hunk 17]
```diff
@@ -0,0 +1,945 @@
...
+
+    ```json
+    {
+        "siteId": "site123",
+        "itemId": "folder456",
+        "name": "New Folder Name"
+    }
+    ```
+
+??? note updateList
```
Comment (DinithiDiaz) [inline en/docs/reference/connectors/sharepoint-connector/sharepoint-configuration.md:830]:
```suggestion
??? note "updateList"
```

File: en/docs/reference/connectors/sharepoint-connector/sharepoint-configuration.md [hunk 18]
```diff
@@ -0,0 +1,945 @@
...
+    ```json
+    {
+        "siteId": "site123",
+        "listId": "3fa85f64-5717-4562-b3fc-2c963f66afa6",
+        "displayName": "Updated List Name",
+        "description": "This is the updated description for the list."
+    }
+    ```
+
+??? note updateListItemFields
```
Comment (DinithiDiaz) [inline en/docs/reference/connectors/sharepoint-connector/sharepoint-configuration.md:882]:
```suggestion
??? note "updateListItemFields"
```

File: en/docs/reference/connectors/sharepoint-connector/sharepoint-configuration.md [hunk 19]
```diff
@@ -0,0 +1,945 @@
...
+            <th>Required</th>
+        </tr>
+        <tr>
+            <td><code>name</code></td>
+            <td>Unique name to identify the connection.</td>
+            <td>Yes</td>
+        </tr>
+        <tr>
+            <td><code>base</code></td>
+            <td>The service root URL. The default value is `https://graph.microsoft.com/v1.0` .</td>
```
Comment (DinithiDiaz) [inline en/docs/reference/connectors/sharepoint-connector/sharepoint-configuration.md:24]:
```suggestion
            <td>The service root URL. The default value is `https://graph.microsoft.com/v1.0`.</td>
```

File: en/docs/reference/connectors/sharepoint-connector/sharepoint-configuration.md [hunk 20]
```diff
@@ -0,0 +1,945 @@
...
+    {
+        "site_id": "site123",
+        "parent_id": "root",
+        "folder_name": "New Folder",
+        "MicrosoftGraphConflictBehavior": "fail"
+    }
+    ```
+
+??? note "uploadFile"
+    The `uploadFile` operation uploads a new file to the specified folder within a SharePoint site. Upon successful execution, the newly uploaded file is available in the target location.
```
Comment (DinithiDiaz) [inline en/docs/reference/connectors/sharepoint-connector/sharepoint-configuration.md:118]:
```suggestion
    The `uploadFile` operation uploads a new file to the specified folder within a SharePoint site. Upon successful execution, the newly uploaded file will be available at the target location.
```

File: en/docs/reference/connectors/sharepoint-connector/sharepoint-overview.md [hunk 21]
```diff
@@ -0,0 +1,29 @@
...
+|-------------------|-------------|
+| 1.1.0             | MI 4.3.0 |
+
+## Sharepoint Connector documentation
+
+* **[Sharepoint Connector Reference]({{base_path}}/reference/connectors/sharepoint-connector/sharepoint-configuration/)**: This documentation provides a reference guide for the Sharepoint Connector.
+
+## How to contribute
+
+As an open source project, WSO2 extensions welcome contributions from the community.
```
Comment (DinithiDiaz) [inline en/docs/reference/connectors/sharepoint-connector/sharepoint-overview.md:23]:
```suggestion
As an open-source project, WSO2 extensions welcome contributions from the community. 
```

**Comment by aashikam** [comment]
@DinithiDiaz Hey small suggestion, when you commit suggestions add them to a batch and commit as one. So there will be only one commit with all the changes. Otherwise there will be redundant commit messages for small changes. 

Screenshot 2024-11-12 at 11 44 44

Screenshot 2024-11-12 at 11 44 53

cc: @arunans23

File: en/docs/assets/img/integrate/connectors/sharepoint-store.png
Comment (aashikam) [reply en/docs/assets/img/integrate/connectors/sharepoint-store.png:1]: 0e51a8f8ffc8d7555b702aa27e6141f2dc878dc0

**Review by arunans23 — APPROVED** [review]

**Pull request merged**